Sources/FoundationEssentials/Calendar/Calendar_Gregorian.swift

```diff
@@ -115,19 +115,26 @@ enum ResolvedDateComponents {
     }
 
     init(dateComponents components: DateComponents) {
-        var (year, month) = Self.yearMonth(forDateComponent: components)
+        let (year, month) = Self.yearMonthResolvingEra(forDateComponent: components)
         let minWeekdayOrdinal = 1
 
         // TODO: Check day of year value here
         if let d = components.day {
-            if components.yearForWeekOfYear != nil, let weekOfYear = components.weekOfYear {
-                if components.month == nil && weekOfYear >= 52 {
-                    year += 1
-                } else if weekOfYear == 1 {
-                    year -= 1
+            let extendedYear: Int
```

Comment:
Is this really 'yearForWeekOfYear'?

Reply:
Are you referring to the line below? 
```swift
if components.yearForWeekOfYear != nil, let weekOfYear = components.weekOfYear,
```

Here we're checking if `weekOfYear` is set. And if it is, we would also expect `yearForWeekOfYear` to be set instead of `year`, don't we?

Reply:
All I meant here was that the concept of an "extended year" is the same as a "year for week of year", right? If so I was wondering if the new terminology may be confusing to readers.

Reply:
I borrowed that term from ICU. It actually means "either `year` or `yearForWeekOfYear`, depending on if you want to calculate a date from given `month`+`day`,  or `week`+ `weekday`"

This isn't very clear admittedly, so I'm still thinking about refactoring this to make it more obvious.